src/builtins.ts

```diff
@@ -8691,134 +8688,184 @@ export function compileVisitGlobals(compiler: Compiler): void {
   );
 }
 
-/** Compiles the `visit_members` function. */
-export function compileVisitMembers(compiler: Compiler): void {
+/** Ensures that the visitor function of the specified class is compiled. */
+function ensureVisitMembersOf(compiler: Compiler, instance: Class): void {
```

Comment:
A new runtime visitor implementation that generates per-class functions instead of a huge unwieldy switch. Increases binary size somewhat, but is much easier to reason about.